Functions/Administration/xASL_adm_SortDicomToFolders.m

```diff
@@ -57,50 +57,51 @@ function xASL_adm_SortDicomToFolders(pathDICOM, nDirLayers, bUseDCMTK, bVerbose)
 			% Read the DICOM file - all error handling is inside this function
 			tDcm = xASL_io_DcmtkRead(Flist{iL}, false, bUseDCMTK, true);
 
+			hasProtocolName = isfield(tDcm, 'ProtocolName') && ~isempty(tDcm.ProtocolName);
+			hasSeriesDescription = isfield(tDcm, 'SeriesDescription') && ~isempty(tDcm.SeriesDescription);
+			hasSeriesNumber = isfield(tDcm, 'SeriesNumber') && ~isempty(tDcm.SeriesNumber);
+
 			% Check that the header was read and contains the basic tags
 			if isempty(tDcm)
                 warning(['Empty DICOM header, skipping: ' Flist{iL}]);
-                hasProtocolName = isfield(tDcm, 'ProtocolName') && ~isempty(tDcm.ProtocolName);
-                hasSeriesDescription = isfield(tDcm, 'SeriesDescription') && ~isempty(tDcm.SeriesDescription);
-                hasSeriesNumber = isfield(tDcm, 'SeriesNumber') && ~isempty(tDcm.SeriesNumber);
-            elseif ~hasProtocolName || ~hasSeriesDescription || ~hasSeriesNumber
+			elseif ~hasProtocolName || ~hasSeriesDescription || ~hasSeriesNumber
                 warning(['DICOM header without ProtocolName, SeriesDescription, or SeriesNumber, skipping: ' Flist{iL}]);
-
+			else
                 % Manage directory name
                 % Priority:
                 % 1. ProtocolName
                 % 2. SeriesDescription
                 % 3. SeriesNumber
 
                 Fname = [];
-                foundName = false;
 
                 % Always add the protocol name to the directory name
                 if hasProtocolName
                     Fname = tDcm.ProtocolName;
-                    foundName = true;
                 end
                 
 				% Add series description if available
 				if hasSeriesDescription
-                    if foundName && ~strcmpi(tDcm.ProtocolName, tDcm.SeriesDescription)
+                    if ~isempty(Fname) && ~strcmpi(tDcm.ProtocolName, tDcm.SeriesDescription)
                         % if ProtocolName was available, was append SeriesDescription if it differs from ProtocolName
 					    Fname = [Fname '_' tDcm.SeriesDescription];
                     else
                         % Only SeriesDescription is also fine
-                        foundName = true;
                         Fname = tDcm.SeriesDescription;
                     end
 				end                
 
-                if ~foundName
+                if isempty(Fname)
                     warning(['ProtocolName & SeriesDescription missing: ' Flist{iL}]);
                 end
 
                 % Add SeriesNumber if available
 				if hasSeriesNumber
-					Fname = [Fname '_' xASL_num2str(tDcm.SeriesNumber)];
-                end
+					if ~isempty(Fname)
```

Comment:
You also want to add seriesNumber if none of the other fields were found?

Reply:
Yes of course... It is added. I just avoid adding the underscore unless there is something in the string already.

So the functionality isn't changed. SeriesNumber is added, we just remove here the leading underscore.